File: src/main/java/org/goafabric/personservice/persistence/entity/TenantAware.java
package org.goafabric.personservice.persistence.entity;

import jakarta.persistence.MappedSuperclass;
import org.hibernate.annotations.Filter;
import org.hibernate.annotations.FilterDef;
import org.hibernate.annotations.ParamDef;

@FilterDef(name = "tenantFilter", parameters = @ParamDef(name = "tenantId", type = String.class))
@Filter(name = "tenantFilter", condition = "tenant_id = :tenantId")
@MappedSuperclass
public class TenantAware {
}

File: src/main/java/org/goafabric/personservice/persistence/multitenancy/TenantAware.kt
package org.goafabric.personservice.persistence.multitenancy

import org.goafabric.personservice.crossfunctional.HttpInterceptor
import org.goafabric.personservice.persistence.audit.AuditJpaListener
import jakarta.persistence.Access
import jakarta.persistence.AccessType
import jakarta.persistence.EntityListeners
import jakarta.persistence.MappedSuperclass

@MappedSuperclass
@EntityListeners(AuditJpaListener::class)
abstract class TenantAware {
    companion object {
        const val TENANT_FILTER = "TENANT_FILTER"
    }

    @Access(AccessType.PROPERTY)
    fun getTenantId(): String? { return HttpInterceptor.getTenantId() }

    fun setTenantId(tenantId: String?) {}

    abstract fun getMyId() : String
}